src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<ExpressionSyntax>
+        {
+            private readonly ArrayBuilder<ExpressionSyntax> _assignmentTargets;
+
+            private Rewriter()
+            {
+                _assignmentTargets = ArrayBuilder<ExpressionSyntax>.GetInstance();
+            }
+
+            private void Free()
+            {
+                _assignmentTargets.Free();
```

Comment:
a bit odd to release some resources, but not hte rewriter itself.  consider pooling these if you care.